src/m365/teams/commands/report/report-pstncalls.spec.ts

```diff
@@ -0,0 +1,260 @@
+      options: {
+        fromDateTime: '2020-12-01',
+        toDateTime: 'abc'
+      }
+    });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation on number of days between fromDateTime and toDateTme exceeding 90', () => {
+    const actual = command.validate({
+      options: {
+        fromDateTime: '2020-08-01',
+        toDateTime: '2020-12-01'
+      }
+    });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation on valid fromDateTime', () => {
+    const actual = command.validate({
+      options: {
+        fromDateTime: '2020-12-01'
+      }
+    });
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation on valid fromDateTime and toDateTime', () => {
+    const actual = command.validate({
+      options: {
+        fromDateTime: '2020-11-01',
+        toDateTime: '2020-12-01'
+      }
+    });
+    assert.strictEqual(actual, true);
+  });
+
+  it('get pstncalls in teams', (done) => {
+    const requestStub: sinon.SinonStub = sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/communications/callRecords/getPstnCalls(fromDateTime=2019-11-01,toDateTime=2019-12-01)`) {
+        return Promise.resolve(jsonOutput);
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, { options: { debug: false, fromDateTime: '2019-11-01', toDateTime: '2019-12-01' } }, () => {
+      try {
+        assert.strictEqual(requestStub.lastCall.args[0].url, "https://graph.microsoft.com/beta/communications/callRecords/getPstnCalls(fromDateTime=2019-11-01,toDateTime=2019-12-01)");
+        assert.strictEqual(requestStub.lastCall.args[0].headers["accept"], 'application/json;odata.metadata=none');
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('get pstncalls in teams with no toDateTime specified', (done) => {
+
+    const now = new Date();
+    sinon.useFakeTimers(now);
+    const toDateTime: string = encodeURIComponent(now.toISOString());
+
+    const requestStub: sinon.SinonStub = sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/communications/callRecords/getPstnCalls(fromDateTime=2019-11-01,toDateTime=${toDateTime})`) {
+        return Promise.resolve(jsonOutput);
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, { options: { debug: false, fromDateTime: '2019-11-01' } }, () => {
+      try {
+        assert.strictEqual(requestStub.lastCall.args[0].url, `https://graph.microsoft.com/beta/communications/callRecords/getPstnCalls(fromDateTime=2019-11-01,toDateTime=${toDateTime})`);
+        assert.strictEqual(requestStub.lastCall.args[0].headers["accept"], 'application/json;odata.metadata=none');
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('get pstncalls in teams output in txt', (done) => {
```

Comment:
Since output type is handled centrally, there is no need to include output-specific tests in commands